### PR TITLE
Added getdaft as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/help-rust it looks like rust is failing to build openssl, though the error message isn't helpful to me, something about a missing file?

File: recipes/getdaft/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+  host:
+    - python
+    - maturin >=1.2.0,<1.3.0
+    - pip
```
Comment (xhochy) [inline recipes/getdaft/meta.yaml:24]:
```suggestion
    - pip
    - openssl
```

File: recipes/getdaft/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/getdaft-{{ version }}.tar.gz
+  sha256: 2916687428b39b6a35f810983ba333e2b418d1905c1b7d5baa0f2c59beb5da1d
+
+build:
+  skip: true  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```
Comment (xhochy) [inline recipes/getdaft/meta.yaml:14]:
```suggestion
  script: 
    - export OPENSSL_DIR=${PREFIX}
    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/getdaft/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+  run:
+    - python
+    - pyarrow >=6.0.1
+    - fsspec
+    - psutil
+    - tqdm
+    - typing-extensions >=4.0.0  # [py<310]
+    - pickle5 >=0.0.12  # [py<38]
+
```
Comment (xhochy) [inline recipes/getdaft/meta.yaml:33]:
```suggestion
```

**Review by xhochy — COMMENTED** [review]
This should help

**Comment by thewchan** [comment]
@xhochy no luck I think; same error

File: recipes/getdaft/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/getdaft-{{ version }}.tar.gz
+  sha256: 2916687428b39b6a35f810983ba333e2b418d1905c1b7d5baa0f2c59beb5da1d
+
+build:
+  skip: true  # [py<37]
+  script:
+    - export OPENSSL_DIR=${PREFIX}
```
Comment (xhochy) [inline recipes/getdaft/meta.yaml:15]:
```suggestion
    - export OPENSSL_DIR=${PREFIX}
    - export OPENSSL_NO_VENDOR=1
```

**Comment by xhochy** [comment]
Got this further now but for this error, I also have no idea.

**Comment by thewchan** [comment]
@xhochy seems to be related to this: https://users.rust-lang.org/t/error-e0635-unknown-feature-stdsimd/106445

File: recipes/getdaft/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    - export OPENSSL_NO_VENDOR=1
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
```
Comment (xhochy) [inline recipes/getdaft/meta.yaml:23]:
```suggestion
    - {{ compiler('rust') }} 1.77
```

**Comment by thewchan** [comment]
@xhochy seems like this is good to go now? Would you like to be a maintainer as well or nah?

**Comment by xhochy** [comment]
> @xhochy seems like this is good to go now? Would you like to be a maintainer as well or nah?

No, but feel free to highlight me if it breaks again.

**Pull request merged**